crates/s3s/src/http/body.rs

```diff
@@ -153,7 +159,14 @@ impl Stream for Body {
     type Item = Result<Bytes, StdError>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        http_body::Body::poll_data(self, cx)
+        match http_body::Body::poll_frame(self, cx) {
+            Poll::Ready(Some(Ok(f))) if f.is_data() => Poll::Ready(Some(Ok(f.into_data().expect("already checked")))),
+            Poll::Ready(Some(Ok(f))) if f.is_trailers() => Poll::Ready(None),
+            Poll::Ready(Some(Ok(_))) => Poll::Ready(Some(Err("unexpected BodyFrame".into()))),
+            Poll::Ready(Some(Err(err))) => Poll::Ready(Some(Err(err))),
+            Poll::Ready(None) => Poll::Ready(None),
+            Poll::Pending => Poll::Pending,
+        }
```

Comment:
how should we handle non Data frames?

Reply:
We just discard non-Data frames. In future refactors, we may add a stream adapter instead of implementing Stream for Body.